apps/storybook/stories/react-essentials/useUpdateLoop.stories.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Column,
+  IconButton,
+  PauseIcon,
+  PlayArrowIcon,
+  Row,
+  ShapedBox,
+  StopIcon,
+  background,
+  shape,
+  size,
+} from "@codedazur/react-components";
+import { Frame, useUpdateLoop } from "@codedazur/react-essentials";
+import { Meta, StoryObj } from "@storybook/react";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { DebugOverlay } from "../../components/DebugOverlay";
+import docs from "./useUpdateLoop.docs.mdx";
+import styled from "styled-components";
+
+interface UseUpateLoopArgs {
+  timeScale?: number;
+  targetFps?: number;
+}
+
+const meta: Meta<UseUpateLoopArgs> = {
+  title: "react-essentials/useUpdateLoop",
+  parameters: {
+    docs: {
+      page: docs,
+    },
+  },
+  argTypes: {
+    timeScale: { control: { type: "range", min: 0, max: 2, step: 0.1 } },
+    targetFps: { control: { type: "number", min: 0.1, max: 240 } },
+  },
+  args: {
+    timeScale: 1,
+    targetFps: undefined,
+  },
+};
+
+export default meta;
+
+export const Default: StoryObj<UseUpateLoopArgs> = {
+  render: (args) => {
+    const [frame, setFrame] = useState<Frame>();
+
+    const { status, start, pause, stop, isUpdating } = useUpdateLoop({
+      onUpdate: setFrame,
+      ...args,
```

Comment:
i feel like we should avoid destructing here to make it easier to understand, wdyt?